pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -689,6 +704,10 @@ stdenv.mkDerivation ({
         else "${llvmPackages.clang}/bin/${llvmPackages.clang.targetPrefix}clang"
       }"
   ''
+  + lib.optionalString (stdenv.hostPlatform != stdenv.targetPlatform && stdenv.targetPlatform.isWindows) ''
```

Comment:
Why did you make this conditional on cross?

Reply:
Hmm I guess I wrongly interpreted the error as caused by the settings from linux to be used by a ghc that compiles to windows but is due to the more general problem of the build config leaking and cross is just a specific case of that. 
In which case it makes sense to have this always happen when `targetPlatform.isWindows`